hadoop-hdds/rocksdb-checkpoint-differ/src/main/java/org/apache/ozone/rocksdiff/RocksDBCheckpointDiffer.java

```diff
@@ -471,14 +524,137 @@ public HashSet<String> readRocksDBLiveFiles(String dbPathArg) {
     return liveFiles;
   }
 
-  // Given the src and destination Snapshots, it prints a Diff list.
-  private synchronized void printSnapdiffSSTFiles(
-      Snapshot src, Snapshot dest) throws RocksDBException {
-    LOG.warn("Src Snapshot files :" + src.dbPath);
+  private int getSnapshotIdx(Snapshot src) {
+    // Jump an O(n) search (for the UT)
+    for (int i = 0; i < allSnapshots.length; i++) {
+      if (allSnapshots[i].snapshotID.equals(src.snapshotID)) {
+        return i;
+      }
+    }
+    return -1;
+  }
+
+  private String[] inputFilesRead;
+
+  /**
+   * Process each line of compaction log text file input and populate the DAG.
+   * TODO: Drop synchronized? and make this thread safe?
```

Comment:
is  processCompactionLogLine() only synchronized to protect the inputFilesRead field?

Reply:
oh i see it's synchronized to protect the dags.